qiskit/quantum_info/states/measures.py

```diff
@@ -252,3 +252,41 @@ def entanglement_of_formation(state):
     conc = concurrence(state)
     val = (1 + np.sqrt(1 - (conc**2))) / 2
     return shannon_entropy([val, 1 - val])
+
+
+def negativity(state, qargs, base=2):
+    r"""Calculates the negativity and logarithmic negativity
+
+    The mathematical expression for negativity is given by:
+    .. math::
+        {\cal{N}}(\rho) = \frac{|| \rho^{T_A}|| - 1 }{2}
+
+    The mathematical expression for logarithmic negaitivity is given by:
+    .. math::
+        {\cal{E}}_N(\rho) = \log_2(|| \rho^{T_A}||)
+
+    Args:
+        state (Statevector or DensityMatrix): a quantum state.
+        qargs (list): The subsystems to be transposed.
+        base (float): the base of the logarithm [Default: 2].
+
+    Returns:
+        negv (float): Negativity value of the quantum state
+        log_negv (float): Logarithmic Negativity value of the quantum state
+
+    Raises:
+        QiskitError: if the input state is not a valid QuantumState.
+    """
+    import math
+
+    if isinstance(state, Statevector):
+        # If input is statevector then converting it into density matrix
+        state = DensityMatrix(state)
+    # Generating partially transposed state
+    state = state.partial_transpose(qargs)
+    eigvals, _ = np.linalg.eig(np.matmul(np.transpose(np.conjugate(state.data)), state.data))
+    eigvals = np.sum(np.sqrt(np.maximum(np.real(eigvals), 0.0)))
+
+    negv = (eigvals - 1) / 2
+    log_negv = math.log(eigvals, base)
+    return negv, log_negv
```

Comment:
Returning both values is unnecessary. Could you return only `negv`?  If you think logarithmic negativity is necessary, please make another function `logarithmic_negativity`.

Reply:
I had a quick suggestion; not sure what you think. Given that the logarithmic negativity is so closely related to negativity (LN(ρ) = log₂(2 N(ρ) + 1)), maybe it would be good if there was just a parameter in this function that defaults to `False`, but when set to `True` it returns `log_negv` instead of `negv`. That way you don't need a completely separate function to just calculate a simple one line of code, although I'm not sure if that's what is preferred within qiskit.

Also, it seems that in the [original paper](https://arxiv.org/pdf/quant-ph/0505071.pdf#page4) the definition is LN(ρ) = log₂(||ρᵀ||₁), but here you're using `math.log`, which is the natural log by default. Lastly, since `numpy` is already imported for other purposes, you could use `np.log` instead of `math.log` and that saves you from importing `math`.